Add spark connect limitation on streaming (#763)
spiceai/docs: website/docs/components/data-connectors/spark.md

## Limitations

> - Correlated scalar subqueries can only be used in filters, aggregations, projections, and UPDATE/MERGE/DELETE commands. [Spark Docs](https://spark.apache.org/docs/latest/sql-error-conditions-unsupported-subquery-expression-category-error-class.html#unsupported_correlated_scalar_subquery)

Reviewer: ```suggestion
- Correlated scalar subqueries are only supported in filters, aggregations, projections, and UPDATE/MERGE/DELETE commands. [Spark Docs](https://spark.apache.org/docs/latest/sql-error-conditions-unsupported-subquery-expression-category-error-class.html#unsupported_correlated_scalar_subquery)
```

> - Spark connector does not support streaming data when reading / performing queries from the connector.

Reviewer: ```suggestion
- The Databricks Catalog Connector (`mode: spark_connect`) does not yet support streaming query results from Spark.
```
